ledger/internal/evaldebugger.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package internal
+
+import (
+	"fmt"
+
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+)
+
+// ProcessTransactionGroupForDebugger ..
+func (eval *BlockEvaluator) ProcessTransactionGroupForDebugger(group []transactions.SignedTxnWithAD) (ledgercore.StateDelta, []transactions.SignedTxnInBlock, error) {
```

Comment:
@jdtzmn Request for clarification with the caveat that I'm unsure how broad of a topic I'm broaching:  Is it intentional for `ProcessTransactionGroupForDebugger` to duplicate `ProcessBlockForIndexer`?

* At first glance, it seems like `ProcessBlockForIndexer` can be reused.  
  * Possibly, `ProcessBlockForIndexer` can be renamed too?  I see no indexer specific behavior though there may be historical reasons in play.
* My unsubstantiated guess is that you created a new method to avoid any semantics that _may_ be associated with `ProcessBlockForIndexer`.

Reply:
@jdtzmn Sorry, disregard the question.  The signatures vary:  `Block` vs `[]transactions.SignedTxnWithAD`.

Though I still think I don't totally grok if necessity, convenience, or something else points to the unique method.

Reply:
@jdtzmn Resolving - Let's pick up the thread in https://github.com/algorand/go-algorand/pull/4322#discussion_r939093906.

https://github.com/algorand/go-algorand/pull/4322#discussion_r939093906 asks a broader question more directly than I'm formulating here.